src/spaniel-observer.ts

```diff
@@ -134,15 +132,25 @@ export class SpanielObserver implements SpanielObserverInterface {
       this.queuedEntries = [];
     }
   }
-  private generateSpanielEntry(entry: IntersectionObserverEntry, state: SpanielThresholdState): SpanielObserverEntry {
+  private generateSpanielEntry(
+    entry: MaybeInternalIntersectionObserverEntry,
+    state: SpanielThresholdState
+  ): SpanielObserverEntry {
     let { intersectionRatio, rootBounds, boundingClientRect, intersectionRect, isIntersecting, time, target } = entry;
     let record = this.recordStore[(<SpanielTrackedElement>target).__spanielId];
-    const timeOrigin = w.performance.timeOrigin || w.performance.timing.navigationStart;
-    const unixTime = this.usingNativeIo ? Math.floor(timeOrigin + time) : time;
+    const unixTime = this.usingNativeIo
```

Comment:
This ultimately becomes `visibleTime` in the watcher callback right?

We should stick to `Date.now` for start time (`visibleTime`), and use hr time for calculating the duration. Mixing `timeOrigin` / `timing.navigationStart` and `Date.now` is not reliable.

Pausing the hr time clock during sleep is okay for us to calculate impression duration.

Reply:
if we use `Date.now` when using native intersection observer, that's also lossy because the callback is async. So capturing a `Date.now()` in the callback is not necessarily when the event happened. do you have an opinion on which approach is less bad?

Reply:
Also, I don't think we are ever mixing `timeOrigin` / `timing.navigationStart` and `Date.now` for any individual entry (as in, https://developer.mozilla.org/en-US/docs/Web/API/IntersectionObserverEntry). I would rather use the HR time all throughout the entire entry or Date throughout the entire entry.

Reply:
The line below, `Math.floor(w.performance.timeOrigin || w.performance.timing.navigationStart + time)` is `time` unix time? The correct should be hrTime which is relative to timeOrigin. Logic looks incorrect to me.

Left another related comment at line 209.

Reply:
If the logic is indeed incorrect, do we have a test to cover?

Reply:
> The correct should be hrTime which is relative to timeOrigin
Sort of, it's correct if we want to mirror native IO, but that is a breaking change.

This change just modifies how duration is calculated.

Added tests for visibleTime earlier today: https://github.com/linkedin/spaniel/blob/v2-release/test/playwright/modules/visible-time.ts

Reply:
Ah wait I see what you mean, how using native IO previously meant using hr time. I will add another test for this behavior first.

Reply:
wait no we always are using unix timestamp time for the `time` field, even for native IO: https://github.com/linkedin/spaniel/blob/v2-release/src/spaniel-observer.ts#L141
So we should keep that behavior for now until a breaking change. I'll add a test regardless